Userland/Games/Spider/Game.cpp

```diff
@@ -284,30 +288,59 @@ void Game::mouseup_event(GUI::MouseEvent& event)
         return;
 
     bool rebound = true;
-    for (auto& stack : m_stacks) {
-        if (stack.is_focused())
-            continue;
+    if (event.button() == GUI::MouseButton::Secondary) {
+        // This enables the game to move the focused cards to the first possible stack excluding empty stacks.
+        // NOTE: I decided to ignore empty stacks, as the game has no undo button, and a card, which has been moved to an empty stack without any other possibilities is not reversable.
+        for (auto& stack : m_stacks) {
+            if (stack.is_focused())
+                continue;
+
+            if (stack.is_allowed_to_push(m_focused_cards.at(0), m_focused_cards.size(), Cards::CardStack::Any) && !stack.is_empty()) {
+                for (auto& to_intersect : m_focused_cards) {
+                    mark_intersecting_stacks_dirty(to_intersect);
+                    stack.push(to_intersect);
+                    (void)m_focused_stack->pop();
+                }
 
-        for (auto& focused_card : m_focused_cards) {
-            if (stack.bounding_box().intersects(focused_card.rect())) {
-                if (stack.is_allowed_to_push(m_focused_cards.at(0), m_focused_cards.size(), Cards::CardStack::Any)) {
-                    for (auto& to_intersect : m_focused_cards) {
-                        mark_intersecting_stacks_dirty(to_intersect);
-                        stack.push(to_intersect);
-                        (void)m_focused_stack->pop();
-                    }
+                update_score(-1);
+
+                update(m_focused_stack->bounding_box());
+                update(stack.bounding_box());
+
+                detect_full_stacks();
 
-                    update_score(-1);
+                ensure_top_card_is_visible(*m_focused_stack);
 
-                    update(m_focused_stack->bounding_box());
-                    update(stack.bounding_box());
+                rebound = false;
+                break;
+            }
+        }
+    } else {
+        for (auto& stack : m_stacks) {
+            if (stack.is_focused())
+                continue;
 
-                    detect_full_stacks();
+            for (auto& focused_card : m_focused_cards) {
+                if (stack.bounding_box().intersects(focused_card.rect())) {
+                    if (stack.is_allowed_to_push(m_focused_cards.at(0), m_focused_cards.size(), Cards::CardStack::Any)) {
+                        for (auto& to_intersect : m_focused_cards) {
+                            mark_intersecting_stacks_dirty(to_intersect);
+                            stack.push(to_intersect);
+                            (void)m_focused_stack->pop();
+                        }
```

Comment:
This bit seems to be duplicated, could this be extracted to a separate function or local lambda?

Reply:
Good point, done!